### PR TITLE
feat: get data for proposal list page

### PR DESCRIPTION
**Pull request by bkioshn**
## Describe your changes
Get data for proposal list page

### CODE REVIEW

**Review by songwongtp — APPROVED** [review]

File: src/lib/services/expression/proposalExpression.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import type { ProposalStatus, ProposalType } from "lib/types";
+
+export const useProposalListExpression = (
+  statuses: ProposalStatus[],
+  types: ProposalType[],
+  search: string,
+  isMyProposal: boolean
+) => {
+  const { address } = useWallet();
```
Comment (poomthiti) [inline src/lib/services/expression/proposalExpression.ts:11]: receive as props instead so this wont be a hook

File: src/lib/services/proposalService.ts [hunk 2]
```diff
@@ -159,3 +165,77 @@ export const useProposalsCountByWalletAddress = (
...
+  search: string,
+  isMyProposal: boolean
+): UseQueryResult<Proposal[]> => {
+  const { indexerGraphClient } = useCelatoneApp();
+  const expression = useProposalListExpression(
+    statuses,
+    types,
+    search,
+    isMyProposal
+  );
```
Comment (poomthiti) [inline src/lib/services/proposalService.ts:183]: retrieve address and pass it here instead

File: src/lib/types/proposal.ts [hunk 3]
```diff
@@ -28,4 +28,5 @@ export interface Proposal {
   resolvedHeight: Option<number | null>;
   type: ProposalType;
   proposer: Option<Addr>;
+  isExpedited: Option<boolean | null>;
```
Comment (poomthiti) [inline src/lib/types/proposal.ts:31]: I think its just boolean

File: src/lib/services/proposalService.ts [hunk 4]
```diff
@@ -159,3 +165,77 @@ export const useProposalsCountByWalletAddress = (
...
+export const useProposalList = (
+  offset: number,
+  pageSize: number,
+  statuses: ProposalStatus[],
+  types: ProposalType[],
+  search: string,
+  isMyProposal: boolean
+): UseQueryResult<Proposal[]> => {
+  const { indexerGraphClient } = useCelatoneApp();
+  const expression = useProposalListExpression(
```
Comment (poomthiti) [inline src/lib/services/proposalService.ts:178]: should we memoized this? @evilpeach

**Review by poomthiti — CHANGES_REQUESTED** [review]

File: src/lib/types/proposal.ts (hunk 3 above)
Comment (bkioshn) [reply src/lib/types/proposal.ts:31]: The generated type is `boolean | null | undefined`

File: src/lib/services/expression/proposalExpression.ts (hunk 1 above)
Comment (evilpeach) [reply src/lib/services/expression/proposalExpression.ts:11]: Agreed, just make it a utility function.

File: src/lib/services/proposalService.ts (hunk 4 above)
Comment (evilpeach) [reply src/lib/services/proposalService.ts:178]: You can use `useMemo` after converting this to a utility function.

File: src/lib/services/proposalService.ts [hunk 5]
```diff
@@ -48,7 +48,7 @@ export const useRelatedProposalsByContractAddressPagination = (
           resolvedHeight: proposal.resolved_height,
           type: proposal.proposal.type as ProposalType,
           proposer: proposal.proposal.account?.address as Addr,
-          isExpedited: proposal.proposal.is_expedited,
+          isExpedited: !!proposal.proposal.is_expedited,
```
Comment (poomthiti) [inline src/lib/services/proposalService.ts:51]: using Boolean() might provide better readability

File: src/lib/services/proposalService.ts [hunk 6]
```diff
@@ -124,7 +124,7 @@ export const useProposalsByWalletAddressPagination = (
             proposal.contract_proposals.at(0)?.resolved_height,
           type: proposal.type as ProposalType,
           proposer: walletAddress,
-          isExpedited: proposal.is_expedited,
+          isExpedited: !!proposal.is_expedited,
```
Comment (poomthiti) [inline src/lib/services/proposalService.ts:127]: ditto

File: src/lib/services/proposalService.ts [hunk 7]
```diff
@@ -200,7 +200,7 @@ export const useProposalList = (
             resolvedHeight: proposal.resolved_height,
             type: proposal.type as ProposalType,
             proposer: proposal.account?.address as Addr,
-            isExpedited: proposal.is_expedited,
+            isExpedited: !!proposal.is_expedited,
```
Comment (poomthiti) [inline src/lib/services/proposalService.ts:203]: ditto

**Review by poomthiti — CHANGES_REQUESTED** [review]

**Review by poomthiti — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**